wandb/integration/openai/resolver.py

```diff
@@ -145,5 +180,84 @@ def _request_response_result_to_trace(
             )
             for choice in choices
         ]
-        trace = self.results_to_trace_tree(request, response, results, time_elapsed)
-        return trace
+        metrics = self._get_metrics_to_log(request, response, results, time_elapsed)
+        return metrics
+
+    def _get_usage_metrics(
+        self,
+        response: Response,
+        time_elapsed: float,
+        model_alias_and_cost: Optional[Dict[str, Any]] = None,
+    ) -> Dict[str, Any]:
+        """Gets the usage stats from the response object."""
+        usage_stats = {}
+        if response.get("usage"):
+            usage_stats = response["usage"]
+        usage_stats["elapsed_time"] = time_elapsed
+        if model_alias_and_cost:
+            usage_stats["prompt_cost"] = (
+                model_alias_and_cost["cost"]
+                * usage_stats.get("prompt_tokens", 0)
+                / 1000
+            )
+            usage_stats["completion_cost"] = (
+                model_alias_and_cost["cost"]
+                * usage_stats.get("completion_tokens", 0)
+                / 1000
+            )
+            usage_stats["total_cost"] = (
+                usage_stats["prompt_cost"] + usage_stats["completion_cost"]
+            )
+        return usage_stats
+
+    def _get_metrics_to_log(
+        self,
+        request: Dict[str, Any],
+        response: Response,
+        results: List[Any],
+        time_elapsed: float,
+    ) -> Dict[str, Any]:
+        if response["object"] in ("text_completion", "chat.completion"):
+            is_completion = True
+        else:
+            is_completion = False
+        model = response.get("model") or request.get("model")
+        if model:
+            model_alias_and_cost = get_alias_and_cost(
+                model.lower(), is_completion=is_completion
+            )
+        else:
+            model_alias_and_cost = None
+
+        model_alias = model_alias_and_cost["alias"] if model_alias_and_cost else None
+
+        metrics = self._get_usage_metrics(response, time_elapsed, model_alias_and_cost)
+
+        usage_table = []
+        for result in results:
+            row = {
+                "request": result.inputs["request"],
+                "response": result.outputs["response"],
+                "model_alias": model_alias,
+                "model": model,
+                "start_time": response["created"],
+                "end_time": response["created"] + time_elapsed,
+                "request_id": response["id"]
+                if response.get("id")
+                else str(uuid.uuid4()),
```

Comment:
why is this necessary? will look legit in the else case, but is actually generated by us.

Reply:
Please do not resolve incomplete conversations.

replaced with None.